Update

temach opened the pull request:
Git commit messages:
Add dialog to help the user to turn on GPS when its off.
Show device marker on the map. Update marker position dynamically.

There is one new class to create the dialog.
There is one new class CurrentUserLocation to track the current location of the user.

 Src/app/build.gradle was changed so that we compile only the google services that we need (maps and location).  Also extra dependency is added "com.android.support:support-v4:24.2.1", becasue it is needed for the Dialog.

temach commented:
Commit  	https://github.com/iamnp/WelcomeToMoscow/pull/4/commits/6ba00f360c767f4f0ce0610124a361e2e50f69f1 changes the layout for activity_main.
New layout:
![2016-10-20_at_12 57 23_screenshot](https://cloud.githubusercontent.com/assets/3957592/19555480/0877e584-96ce-11e6-8ef0-627dc392dad3.png)

iamnp commented on Src/app/build.gradle:
```diff
@@ -25,6 +25,8 @@ dependencies {
         exclude group: 'com.android.support', module: 'support-annotations'
     })
     compile 'com.android.support:appcompat-v7:24.2.1'
```
Update tp 25.0.0

iamnp commented on Src/app/build.gradle:
```diff
@@ -25,6 +25,8 @@ dependencies {
     })
     compile 'com.android.support:appcompat-v7:24.2.1'
-    compile 'com.google.android.gms:play-services:9.6.1'
+    compile 'com.android.support:support-v4:24.2.1'
```
Update tp 25.0.0

iamnp commented on Src/app/src/main/java/newteam2/welcometomoscow/DialogToEnableGPS.java:
```diff
@@ -0,0 +1,35 @@
+        // Use the Builder class for convenient dialog construction
+        AlertDialog.Builder builder = new AlertDialog.Builder(getActivity());
+        // Question to display
+        builder.setMessage("This app works better with GPS enabled. Turn on GPS now?");
```
Use localizable strings resource

iamnp commented on Src/app/src/main/java/newteam2/welcometomoscow/DialogToEnableGPS.java:
```diff
@@ -0,0 +1,35 @@
+        // Question to display
+        builder.setMessage("This app works better with GPS enabled. Turn on GPS now?");
+        // User accepted the dialog
+        builder.setPositiveButton("yes", new DialogInterface.OnClickListener() {
```
Use localizable strings resource

iamnp commented on Src/app/src/main/java/newteam2/welcometomoscow/MapsActivity.java:
```diff
@@ -26,6 +27,19 @@ protected void onCreate(Bundle savedInstanceState) {
+        // check if enabled and if not send user to the GSP settings
+        // display a dialog and suggest to go to the settings, to turn it on
+        if (!enabled) {
+            DialogFragment d = new DialogToEnableGPS();
```
Consider using final modifier

iamnp commented:
Update Android studio to 2.2.2
Update Android Gradle plugin to 2.2.2
Update Gradle to 2.14.1

temach commented:
I have fixed what was wrong in commit https://github.com/temach/WelcomeToMoscow/commit/6ba00f360c767f4f0ce0610124a361e2e50f69f1 :
1) Gradle is targeting 25.0.0 sdk tools.
2) Strings from code are moved into strings.xml

As for the layout, maybe we can just add it as it is?  
It is ugly and unfunctional, but I need to have at least one "choose this quest" button for map testing. Design can be changed later by Sergey. Personally I don't think it will be so hard to change the design later.
Changing bad code is difficult, but changing bad design is easier.

iamnp commented on Src/app/src/main/java/newteam2/welcometomoscow/QuestInfo.java:
```diff
@@ -0,0 +1,26 @@
+ *  All special images/ all text.
+ *  All time/place values.
+ */
+class QuestInfo {
```
Should have at least one more field: long id

iamnp commented on Src/app/src/main/java/newteam2/welcometomoscow/QuestMenuEntryAdapter.java:
```diff
@@ -0,0 +1,42 @@
+ * Created by artem on 10/20/16.
+ */
+
+public class QuestMenuEntryAdapter extends ArrayAdapter<QuestInfo> {
```
The class should be called "QuestListAdapter" because it is not a menu adapter

iamnp commented on Src/app/src/main/res/layout/activity_main.xml:
```diff
@@ -13,17 +13,39 @@
+            android:onClick="ButtonClickViewMap" />
+    </LinearLayout>
+
+    <ListView
```
Consider using RecyclerView with reusable layouts (https://developer.android.com/training/material/lists-cards.html)

temach replied:
I hope Sergey will code this. I just need a GUI to test the Google maps code.

temach commented:
Updated by branch from your branch (now we can stop using --no-ff option).
Fixed QuestInfo long field. Fixed change of name for QuestListAdapter. 
Using RecyclerView is question for Sergey.
